cloudflare.go

```diff
@@ -341,6 +339,32 @@ func (api *API) makeRequestWithAuthTypeAndHeaders(ctx context.Context, method, u
 		}
 	}
 
+	// Check the Content-Type: if it's multipart then get the first part
+	// TODO: return a proper structure from this method with file names, etc.
+	// TODO: if the content-type is JSON then parse it here and return the parsed structure?
+	mediaType, mediaParams, _ := mime.ParseMediaType(resp.Header.Get("content-type"))
+	if strings.HasPrefix(mediaType, "multipart/") {
+		reader := multipart.NewReader(resp.Body, mediaParams["boundary"])
+		part, err := reader.NextPart()
+		if err != nil {
+			return nil, fmt.Errorf("could not get multipart response body: %w", err)
+		}
+		respBody, err = ioutil.ReadAll(part)
+		if err != nil {
+			return nil, fmt.Errorf("could not read multipart response body: %w", err)
+		}
+	} else {
+		respBody, err = ioutil.ReadAll(resp.Body)
+		defer resp.Body.Close()
+		if err != nil {
+			return nil, fmt.Errorf("could not read response body: %w", err)
+		}
+	}
+
+	if api.Debug {
```

Comment:
this probably shouldn't move as the debug is intended to also allow seeing non-200 HTTP responses to help problem diagnosis.

Reply:
Moved this back where it was and moved the multipart parsing to DownloadWorker itself. Check out this new approach and let me know what you think!

Reply:
ps ... what's going on in cloudflare_experimental.go? It looks like a subtly different API client implementation and is only used in the Zone code.